### PR TITLE
Migrate to ESLint 9 and flat config

### PR DESCRIPTION
**Pull request by philippfromme**
Related to https://github.com/bpmn-io/internal-docs/issues/1042

### Proposed Changes

* updates ESLint to v9
* updates eslint-plugin-bpmn-io to v2
* removes eslint-plugin-import
* fixes newly reported lint errors

 

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [ ] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

**Comment by barmac** [comment]
The [nested test](https://github.com/bpmn-io/diagram-js/blob/develop/test/spec/features/context-pad/ContextPadSpec.js#L1479) looks like a false positive. We should change the function name only.

**Comment by philippfromme** [comment]
So I guess we want to fix things like `no-identical-title` and not disable that rule? We have the same issue in the Camunda Modeler where we would have to fix a couple of things.

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import bpmnIoPlugin from 'eslint-plugin-bpmn-io';
+
+export default [
+  ...bpmnIoPlugin.configs.node.map(config => {
+
+    return {
+      ...config,
+      files: [
+        '*.js',
+        '*.mjs'
```
[the remark this answers is not in the document]
Comment (nikku) [reply eslint.config.mjs:10]: The purpose of this config was to match root level files only :heavy_check_mark:

File: test/spec/features/overlays/OverlaysSpec.js [hunk 2]
```diff
@@ -490,7 +490,7 @@ describe('features/overlays', function() {
     }
 
 
-    it('should position top left of shape', inject(function(overlays) {
+    it(' (connection)', inject(function(overlays) {
```
Comment (nikku) [inline test/spec/features/overlays/OverlaysSpec.js:493]: Is this an accident?

File: test/spec/features/resize/ResizeSpec.js [hunk 3]
```diff
@@ -313,29 +313,6 @@ describe('features/resize - Resize', function() {
     }));
 
 
-    it('should resize to minimum bounds', inject(function(canvas, resize, dragging, elementFactory) {
```
Comment (nikku) [inline test/spec/features/resize/ResizeSpec.js:316]: Removed by accident, or duplicate?

File: test/spec/features/resize/ResizeUtilSpec.js [hunk 4]
```diff
@@ -329,61 +329,6 @@ describe('resize/ResizeUtil', function() {
   });
 
 
-  describe('getMinResizeBounds', function() {
```
Comment (nikku) [inline test/spec/features/resize/ResizeUtilSpec.js:332]: Removed by accident or duplicate?

File: test/spec/i18n/translate/translateSpec.js [hunk 5]
```diff
@@ -38,11 +38,6 @@ describe('i18n - translate', function() {
       expect(translate('<div />')).to.eql('<div />');
     }));
 
-
-    it('should handle missing replacement', inject(function(translate) {
```
Comment (nikku) [inline test/spec/i18n/translate/translateSpec.js:42]: Duplicate :heavy_check_mark:

**Comment by nikku** [comment]
@philippfromme Updated this PR as discussed with proper configuration of browser vs. node environments (https://github.com/bpmn-io/diagram-js/pull/936/commits/ce60276a25f53cadd32397b5d552e92dbf7ee355). After you do it, then you see that https://github.com/bpmn-io/diagram-js/pull/936/commits/85027fef783bf52e20cd49d94d8f36a24a26ff39 is indeed necessary. No-one but us defines that global.

**Comment by nikku** [comment]
Updated to designated eslint format: https://github.com/bpmn-io/diagram-js/pull/936/commits/4b179558d18867c8d8b7da26f22c00f985dc1aac.

**Comment by nikku** [comment]
@philippfromme Some open questions on this PR.

**Comment by philippfromme** [comment]
@nikku The deleted tests were duplicates.

**Pull request merged**